tools/make.py

```diff
@@ -228,57 +228,31 @@ def find_bi_tools(work_drive):
         raise Exception("BadTools","Arma 3 Tools are not installed correctly or the P: drive needs to be created.")
 
 
-def find_depbo_tools(regKey):
+def find_depbo_tools():
     """Use registry entries to find DePBO-based tools."""
-    stop = False
-
-    if regKey == "HKCU":
-        reg = winreg.ConnectRegistry(None, winreg.HKEY_CURRENT_USER)
-        stop = True
-    else:
-        reg = winreg.ConnectRegistry(None, winreg.HKEY_LOCAL_MACHINE)
-
-    try:
-        try:
-            k = winreg.OpenKey(reg, r"Software\Wow6432Node\Mikero\pboProject")
-        except FileNotFoundError:
-            k = winreg.OpenKey(reg, r"Software\Mikero\pboProject")
-        try:
-            pboproject_path = winreg.QueryValueEx(k, "exe")[0]
-            winreg.CloseKey(k)
-            print("Found pboproject.")
-        except:
-            print_error("Could not find pboProject.")
+    requiredToolPaths = {"pboProject": None,"rapify": None,"MakePbo": None}
+    failed = False
 
+    for tool in requiredToolPaths:
         try:
-            k = winreg.OpenKey(reg, r"Software\Wow6432Node\Mikero\rapify")
+            k = winreg.OpenKey(winreg.HKEY_CURRENT_USER, r"Software\Mikero\{}".format(tool))
         except FileNotFoundError:
-            k = winreg.OpenKey(reg, r"Software\Mikero\rapify")
+            k = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, r"Software\Mikero\{}".format(tool))
         try:
-            rapify_path = winreg.QueryValueEx(k, "exe")[0]
-            winreg.CloseKey(k)
-            print("Found rapify.")
+            path = winreg.QueryValueEx(k, "exe")[0]
+            #Strip any quotations from the path due to a MikeRo tool bug which leaves a trailing space in some of its registry paths.
+            requiredToolPaths[tool] = path.strip('"')
+            print_green("Found {}.".format(tool))
         except:
-            print_error("Could not find rapify.")
-
-        try:
-            k = winreg.OpenKey(reg, r"Software\Wow6432Node\Mikero\MakePbo")
-        except FileNotFoundError:
-            k = winreg.OpenKey(reg, r"Software\Mikero\MakePbo")
-        try:
-            makepbo_path = winreg.QueryValueEx(k, "exe")[0]
+            print_error("Could not find {}".format(tool))
+            failed = True
```

Comment:
Any reason you aren't just raising the exception here? It's been a while since I've used python in any depth

Reply:
it raises an exception on line 252 for the failure after the 3 checks for the tools are made. It's more for ease of debugging why it won't run as it will tell you exactly what it can/cant find

Reply:
Fair 👍 I can't see any issues with the code, will let @jonpas handle approval to put a second set of eyes over it